internal/storage/storage.go

```diff
@@ -40,7 +33,7 @@ type Memory struct {
 	UserURLs map[string][]int
 }
 
-func (m *Memory) AddURL(url string, user string) (string, error) {
+func (m *Memory) AddURL(ctx context.Context, url string, user string) (string, error) {
```

Comment:
здесь и далее для всех функций memory стораджа.
```suggestion
func (m *Memory) AddURL(_ context.Context, url string, user string) (string, error) {
```
это не влияет на компиляцию, но любой линтер будет ругаться на неиспользуемые переменные.